### PR TITLE
Add encryption mode support

### LINKED ISSUE #12: need to store crypto mode in return value
Currently, we return the ciphertext from AES-CBC. However, if we add other encryption modes in the future we will not be able to distinguish ciphertexts. We should store the mode with the ciphertext so that we can automatically distinguish which decryption mode to use.

### PR DESCRIPTION
**Pull request by enj**
Fixes #12
Enables #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,13 +25,15 @@ socket activation is assumed.
 
 ### Required
 
- * `-command string`: the command to retrieve the key encryption key
+ * `--command string`: the command to retrieve the key encryption key
```
Comment (npmccallum) [inline README.md:28]:
```
$ ./kms -help
Usage of ./kms:
  -command string
    	the command to retrieve the key encryption key
  -endpoint string
    	the listen address (ex. unix:///tmp/kms.sock)
  -timeout duration
    	maximum time to cache KEK locally (default 1h0m0s)
```

Can you clarify the difference between single and double dash in this case?

Comment (enj) [reply README.md:28]: There is none (both do the same exact thing).  I just really prefer `--` (and it matches how `kubectl`, `oc`, etc print their flags).

Comment (enj) [reply README.md:28]: I can see if there is a simple way to override how the flags print so they have double dash.

Comment (enj) [reply README.md:28]: #20 to track

**Comment by npmccallum** [comment]
AFAICS, the project won't compile in between the first and second commit. Commits should always compile (and pass tests). I think the first change should do this:
1. Extract interface.
2. Put AESCBC code into a module.
3. Update references so that the code builds.

**Comment by enj** [comment]
@npmccallum pretty sure you are just asking me to squash the first two commits.  I purposefully did not do that because I wanted the git diffs to stay small.

**Comment by npmccallum** [comment]
Except that the cost of doing that is having a bisect that can't work. That's a very high price for a file rename commit (which is not necessary in git).

**Review by npmccallum — COMMENTED** [review]
I'm pretty sure that if you do a git mv and then do your edits the diff will be much smaller.

**Review by npmccallum — APPROVED** [review]

**Pull request merged**